plugins/java-decompiler/engine/src/org/jetbrains/java/decompiler/struct/StructClass.java

```diff
@@ -162,6 +164,24 @@ public ConstantPool getPool() {
     return fields;
   }
 
+  // XXX slow, but no lookup table available
+  public boolean containsFieldWithShortName(String shortName){
+    for(StructField f: fields) {
+      if(shortName.equals(f.getName())) {
```

Comment:
This is actually not about anonymous classes, but about switch generated class:
void foo(DayOfWeek d) {
        switch(d) {
            case FRIDAY: return;
        }
    }